test-poc/framework/src/main/java/org/keycloak/test/framework/database/DatabaseSupplier.java

```diff
@@ -0,0 +1,37 @@
+package org.keycloak.test.framework.database;
+
+import org.keycloak.test.framework.injection.InstanceWrapper;
+import org.keycloak.test.framework.injection.LifeCycle;
+import org.keycloak.test.framework.injection.Registry;
+import org.keycloak.test.framework.injection.Supplier;
+
+import java.lang.annotation.Annotation;
+
+public abstract class DatabaseSupplier implements Supplier<TestDatabase, Annotation> {
```

Comment:
Did you consider some TestManagedDatabase annotation which will be reused within KeycloakIntegrationTest annotation for ServiceLoader injection?

Reply:
e.g. https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/JavascriptBrowser.java#L32